### ISSUE TITLE
Build problems on Ubuntu

### ISSUE THREAD

**Issue by nicklasaven** [tangrams/tangram-es#398]
I get 

undefined reference to symbol 'XF86VidModeQueryExtension'

when I try to build on a fresh Ubuntu 15.10 installation.

Here is some more of the output:

Linking CXX executable bin/tangram
/usr/bin/ld: lib/libglfw3.a(x11_init.c.o): undefined reference to symbol 'XF86VidModeQueryExtension'
/usr/lib/x86_64-linux-gnu/libXxf86vm.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/tangram.dir/build.make:149: recipe for target 'bin/tangram' failed
make[3]: *** [bin/tangram] Error 1
make[3]: Leaving directory '/home/nicklas/dev/tangram-es/build/linux'
CMakeFiles/Makefile2:66: recipe for target 'CMakeFiles/tangram.dir/all' failed
make[2]: *** [CMakeFiles/tangram.dir/all] Error 2
make[2]: Leaving directory '/home/nicklas/dev/tangram-es/build/linux'
Makefile:116: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/nicklas/dev/tangram-es/build/linux'
Makefile:209: recipe for target 'linux' failed
make: *** [linux] Error 2

Google seems to point me in the direction of build flags and not missing dependencies. 

Thanks
Nicklas Avén

**Comment by tallytalwar** [comment]
Hey Nicklas

I tried to build on 15.10, using g++-4.9.3 and it did build fine for me.

Can you let us know what gcc/g++ version you are using? Though it does not seem to be a compiler version error, but just for sanity check, I want to repro the issue with your exact configuration.

Thanks

**Comment by nicklasaven** [comment]
Hi

Thanks for looking at this!

I have:
 gcc version 5.2.1 20151010 (Ubuntu 5.2.1-22ubuntu2)

But I think you are right about some conflict. 
I followed the readme and ran 
git submodule init && git submodule update

But when I tried to build I had dependency problems and istalled what it asked for. I cannot reacall exactly what it asked for but I then installed:
it asked for xrandr so I installed:
libxrandr-dev
then it asked for xinerama and I installed:
libxinerama-dev
it asked for xcursor and i installed:
libxcursor-dev
then I had theerror described above and tried by installing a bunch of libraries:
mesa-common-dev mesa-utils-extra, libgl1-mesa-dev, libglapi-mesa libglu1-mesa-dev

So I guess i messed things up :-)

You ask about glfw. As far as i understand I haven't installed anything like that.
aptitude search glfw gives no reult that is installed. Only uninstalled packages.

Thanks again

**Comment by tallytalwar** [comment]
Thanks for the details. 

I can give it a shot with g++ 5.2 also to make sure its not that.

Can you do a clean install and update us with the exact conflicts you were getting? 

It will be good for us to know and document if ubuntu or other linux distributions ask for other dependencies other than the ones we are handling in our build process.

**Comment by blair1618** [comment]
It seems that GLFW (the windowing library we use) has some dependencies on linux that weren't cited in our build instructions: http://www.glfw.org/docs/latest/compile.html

The README is updated now, but see if installing `xorg-dev` and `libgl1-mesa-dev` helps with the build process.

**Comment by nicklasaven** [comment]
I had the chance today to install Ubuntu on a computor at work. Then I followed the readme and it built perfectly.

One difference between the installs is that at the box at home I first installed Android Studio with Ubuntu Make (umake android)

That uses a special repository and installs a lot of things out of my control. So maybe something gets another version then.

But I had the same gcc now, 5.2.1 by just installing build-essential.
I will look some more at my installation at home later today.
